### ISSUE TITLE
Provide downloads and additional links for our publications

### ISSUE THREAD

**Issue by Toxaris** [ps-mr/website#21]
The publication list would be improved if we would link to

 - the official version of our paper
 - the author's version of our paper
 - the homepage of our paper

As a first step, we have to define how we want to store that information. I'm thinking of some combination of default URLs based on the citation key, metadata from the bibtex file, and metadata from the paper homepage, if on our site. Basic idea:

 (1) Find the paper homepage, by default at `/publications/${citation_key}`. Let the bibtex entry override this with an `url` field doesn't end in `.pdf`.

 (2) for the official version, use the `doi` attribute from the bibtex file.

 (3) for the author's version, use `publications/${citation_key}.pdf` by default. Let the bibtex entry override this with an `url` field that ends in `.pdf`.

 (4) If the paper homepage is on our site, set its default metadata based on the preceding rules 2 and 3, and then let the YAML front matter block override these defaults.

Any thoughts?

**Comment by Blaisorblade** [comment]
+1 to this — my own preprints are already somewhat in disarray. But I have a few remarks.

I'm not sure what's the "paper homepage". I see two things, which shouldn't be confused:
- project homepages (which can be associated to one or more papers though), like http://inc-lc.github.io/. They don't replace paper pages, and often don't map to a single paper, and are often named after the project. They might get linked to inside the paper pages.
- paper pages, auto-generated and containing abstracts, bib entries, and further fields, similarly to paper repositories — like [Joshua Dunfield's](http://www.cs.cmu.edu/~joshuad/by-date.html).
  - they might be inline in the publication list (with a show/hide button), or have a generated URL using the citation key.
  - lots of fields are fixed and should be extracted from the BibTeX file. Since people move, putting more metadata into BibTeX might be good.
  - it'd be cool to get such things without even creating an empty page, if technically possible. Otherwise, we can auto-create files with fixed content (say, a YAML front-matter setting the `paper-page` template, or even empty).
  - it'd also be cool to allow for more content in markdown — in Joshua Dunfield's page, even the links to paper versions, slides, etc. aren't so standard, and this *adds* value to them.

But at least the abstract does not need to be on its own page — they might just be hidden on the publication list. If those exist, the citation key might be reasonable as part of the URL. See http://www.cs.cmu.edu/~joshuad/by-date.html (and linked pages) for one example of doing this right. If we do those, I suspect they should exist somewhat unconditionally (if we have at least an abstract). Failing that, a page [close to this](http://www.cs.cmu.edu/~joshuad/papers/sectanno/) 

For the official version: as a user, I have a minor dislike doi links because of the extra redirect.

On the `url` field: let's please please please have different fields instead of using URLs for two different things.
- We don't need it — BibTeX allows arbitrary fields
- It's fragile (PDF URLs aren't guaranteed to end in `.pdf`, we might want to specify both): the real type is `(Option URL, Option URL)`, not `Option (Either URL URL)`. We might have defaults, but being able to disable them seems important.

We should either use `url` for the actual PDF, if that's standard (er, it's at least my standard), or not use `url` at all (if current usage is inconsistent).

Further thoughts:
- paper pages last longer than affiliations, so putting as much data as possible into BibTeX is probably a good idea. I would avoid Markdown in there, though. Arbitrary fields could be annoying, but auto-renaming them for the transition (if needed) would be rather OK.

**Comment by Toxaris** [comment]
By "paper homepage", I mean a page that contains additional info about the paper (such as downloadable material, the slides, a video, whatever). I think there should be at most one such page per paper.

Technically, we could autogenerate them, but I think we shouldn't, because for some papers, that page should not be on our website (but on some external collaborator's website).

> For the official version: as a user, I have a minor dislike doi links because of the extra redirect.

This redirect is a redirect from a guaranteed-to-be-stable to an potentially unstable URL. I think we should link to the stable URL.

> On the url field: let's please please please have different fields instead of using URLs for two different things.

Ok, can you propose field names?

>  See http://www.cs.cmu.edu/~joshuad/by-date.html (and linked pages) for one example of doing this right.

Getting what right?

**Comment by Blaisorblade** [comment]
> Getting what right?

Sorry, let me rephrase: I looked at what Joshua did and I liked it.

> I propose to first implement linking to existing paper pages, and later, when we have some experience what info we want in a paper page, consider automatic generation again.

Makes sense.

> Design goal: This system should be usable without documentation and without guidance, just by copy-pasting existing data and changing it in the obvious way.

Ah, thanks for explaining that! Now I see why you came up with that design.

Looking at the existing data, it seems that relevant field names should be `pdf` and `url`, that currently both point to PDFs. Moreover, I'll take a look at what bibtex2html (our old tool, and a commonly used one) does. It also seems that the existing data is a mess.

**Comment by Blaisorblade** [comment]
> Design goal: This system should be usable without documentation and without guidance, just by copy-pasting existing data and changing it in the obvious way.

One way to refine that goal could be following bibtex2html as far as convenient, so that entries are reusable:
https://www.lri.fr/~filliatr/bibtex2html/doc/manual.html#sec4

This increases compatibility with existing entries (if they're designed for bibtex2html, and I don't know many standard alternatives; our current bibliography is designed for that) and portability to other systems.

bibtex2html's handling of DOI fields is documented here (but not in the manual):
https://github.com/backtracking/bibtex2html/blob/master/CHANGES

If we use a stricter scheme (which I'd prefer), this already supports:
- official versions with `doi`
- pdfs (like author's versions) with `pdf` links that end in pdf (bibtex2html looks at the suffix, and failing that at the protocol, to determine the link name)
- other URLs (such as the paper page URL) with `url`

We could later add more heuristics to be more compatible with bibtex2html.

I also noticed your proposal allows easily disabling the defaults, by simply not providing the files — so, we cover the whole space of `(Option URL, Option URL)` values. (I had missed that earlier and complained — sorry!).
